### DISCUSSION TITLE
Flicker with NextJS

### DISCUSSION THREAD

**Discussion by konsalex** [KevinVandy/mantine-react-table#225] [Q&A] [answered]
### mantine-react-table version

v1.3.4

### react & react-dom versions

v18.2.0

### Describe the bug and the steps to reproduce it

Visit: https://www.designsystemcookbooks.com/ and try to refresh the page

https://github.com/KevinVandy/mantine-react-table/assets/12672541/a02ea2fb-b80c-4bb6-99ea-e4da3efa82cd

I suspect that this has to do with `emotion` but not sure how to resolve

### Minimal, Reproducible Example - (Optional, but Recommended)

I used a template from NextJS and created a reproduction repo here:
https://github.com/konsalex/reproduction-flicker-mantine

The issue is also here

### Screenshots or Videos (Optional)

https://github.com/KevinVandy/mantine-react-table/assets/12672541/a02ea2fb-b80c-4bb6-99ea-e4da3efa82cd

### Do you intend to try to help solve this bug with your own PR?

No, because I do not know how

### Terms

- [X] I understand that if my bug cannot be reliably reproduced in a debuggable environment, it will probably not be fixed and this issue may even be closed.

**Comment by alessandrojcm** [comment, marked as answer]
Hey @konsalex, yes this is indeed due to Emotion. You have two options here:

* Use MRT v2, you'd need to migrate to Mantine v7 (it uses CSS Modules, you can also use Vanilla Extract or SASS). This is a big breaking change and MRT v2 is in alpha so use it at your own risk.
* You can keep using MRT v1 and take a look at [this](https://github.com/emotion-js/emotion/issues/2928#issuecomment-1780046177) for a workaround.

Let me know how it goes.

**Comment by KevinVandy** [comment]
Follow the Mantine next Js guide
